cmd/influxd/run.go

```diff
@@ -203,14 +203,14 @@ func joinBroker(b *messaging.Broker, joinURLs []*url.URL) {
 }
 
 // creates and initializes a server.
-func openServer(path string, u *url.URL, b *messaging.Broker, initializing, configExists bool, joinURLs []*url.URL, w io.Writer) *influxdb.Server {
+func openServer(authenticationEnabled bool, path string, u *url.URL, b *messaging.Broker, initializing, configExists bool, joinURLs []*url.URL, w io.Writer) *influxdb.Server {
 	// Ignore if there's no existing server and we're not initializing or joining.
 	if !fileExists(path) && !initializing && len(joinURLs) == 0 {
 		return nil
 	}
 
 	// Create and open the server.
-	s := influxdb.NewServer()
+	s := influxdb.NewServer(authenticationEnabled)
```

Comment:
I know we don't want to make it easy to "forget" to enable auth, but I don't think that is likely to happen, especially with testing, but I think this code would be nicer if we simply add:

`(s *Server) EnableAuthenication()`
and
`(s *Server) DsiableAuthenication()`

instead of adding a bool to the constuctor. I don't think it's great style. @dgnorton feels the same, as he has mentioned in previous reviews.